crates/types/src/lib.rs

```diff
@@ -1,12 +1,22 @@
 //! Internal dependency of Wasmtime and Cranelift that defines types for
 //! WebAssembly.
 
-use smallvec::SmallVec;
+#![no_std]
+
+extern crate alloc;
+#[cfg(feature = "std")]
+extern crate std;
```

Comment:
Hoping to learn about `no_std` here: why include this line? The crate should compile and work without it, so is it just here to shadow what gets used in `alloc` with the `std` version instead?

Reply:
Ah this is here for the (one) explicit use of std in this crate, basically just the Error trait implementation. Without it there's no std to pull that trait from

Reply:
Got it, thanks!